Add Developer Certificate of Origin

0xc0170 opened the pull request:
Each commit must have at least one `Signed-off-by:` line from the committer to certify that the contribution is made under the terms of the Developer Certificate of Origin (see dco.txt for more details).

Section Developer’s Certificate of Origin should refer to the full text of DCO, define what is required. The real name and email must be used.
I can add there dco github action (it's part of every pull request).

Part of this, I'll need to set up dco github action to start checking sign offs and also to do a clean up our git history to contain sign off commits (forcing push unfortunately, that should be one time thing once we get this in, we can coordinate so everyone can rebase once done).

pfalcon commented on dco.txt:
```diff
@@ -0,0 +1,37 @@
+Developer Certificate of Origin
```
Should this file be named `DCO.txt`?, to emphasize it's an abbreviation? I have to admit I didn't see many (if at all) such files if other projects, but makes sense to name it similarly to `LICENSE*` `README*`, etc.

pfalcon replied:
Ok, JerryScript has DCO.md: https://github.com/jerryscript-project/jerryscript/blob/master/DCO.md

0xc0170 replied:
I can change to capital as it's an abbreviation.

0xc0170 replied:
The name updated b83bdbf

0xc0170 commented:
I rebased the main history to add sign-off-by notations so we can enable the checks if this goes in: https://github.com/0xc0170/MCU-Driver-HAL/commits/fix-rebased (while adding sign off, I removed merge commits as well so we force push just once if we really want to). Although looking at tags, we already provided 0.1.0. Shall we just keep the history up to today as it is, all commits are from the teams. The history would stay for the first 31 commits as well. 
@pfalcon you were suggesting to possibly rewrite it?

MarceloSalazar commented on CONTRIBUTING.md:
```diff
@@ -26,3 +26,8 @@ either express or implied.
 
+## Developer’s Certificate of Origin
+
+Each commit must have at least one `Signed-off-by:` line from the committer to certify that the contribution is made under the terms of the [Developer Certificate of Origin](./dco.txt).
```
```suggestion
Each commit must have at least one `Signed-off-by:` line from the committer to certify that the contribution is made under the terms of the [Developer Certificate of Origin](./DCO.txt).
```

0xc0170 commented:
I can clean the history once approved.

MarceloSalazar commented:
@0xc0170 please go ahead! :)

evedon reviewed (approved):
Thanks Martin.
Let's go ahead wit this PR and then fix the history. I'd rather that the history is fixed even if we need to force push the tag.

ithinuel commented on CONTRIBUTING.md:
```diff
@@ -26,3 +26,8 @@ either express or implied.
+
+Each commit must have at least one `Signed-off-by:` line from the committer to certify that the contribution is made under the terms of the [Developer Certificate of Origin](./DCO.txt).
+
+If you set your `user.name` and `user.email` as part of your git configuration, you can sign your commit automatically with `git commit -s`. You have to use your real name (i.e., pseudonyms or anonymous contributions cannot be made). This is because the DCO is a legally binding document.
```
This line is misleading as there's not cryptographic signature involved with the use of `git commit -s`.
It literally just adds the `Signed-off-by:` text to your commit message.

Proper digital signature requires the use of `git commit -S` (with a capital `S`). cf [git's documentation](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work) for more information.
"Signed-off-by" is not a stronger proof than what git already has in its meta data with the author and the committer fields.

0xc0170 replied:
that is how it is usually phrased. We can follow what git commit -s help provides, instead of signing:

You can add a signoff to a commit with `git commit -s`

I'll send PR fixing it.

ithinuel commented on DCO.txt:
```diff
@@ -0,0 +1,37 @@
+Developer Certificate of Origin
+Version 1.1
+
+Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
```
Is the work in this repository given to the Linux foundation?

(I am not a lawyer so please pardon my inaccurate wording).

0xc0170 replied:
No, this text is from https://developercertificate.org/. They introduced it.
